src/compiler/data.ts

```diff
@@ -410,6 +415,37 @@ export namespace stack {
   };
 }
 
+export namespace dates {
+  /**
+   * Add data source for with dates for all months, days, hours, ... as needed.
+   */
+  export function defs(model: Model) {
+    let defs = [];
```

Comment:
You never change reference to this array, so this should be `const`.
Another note is that we commonly use `reduce` for this kind of thing elsewhere in this codebase.  Better be consistent.